### ISSUE TITLE
The latest release creates bad containers

### ISSUE THREAD

**Issue by DAKExDUCK** [containrrr/watchtower#1835]
### Describe the bug

Creates new containers for watchtower after last released version of docker image

### Steps to reproduce

- Configured watchtower
- update watchtower to last version of docker image and reload
- see that bug

### Expected behavior

Just update 1 container (watchtower's container)

### Screenshots

![image](https://github.com/containrrr/watchtower/assets/63103528/f7320ec8-4bcb-4c2e-bce8-71f9ed94140e)
![image](https://github.com/containrrr/watchtower/assets/63103528/f89ce275-a554-4de5-8a9e-e0d43cba0257)
![image](https://github.com/containrrr/watchtower/assets/63103528/9f25523d-ebc1-4cda-8b7b-427534067bcf)

### Environment

- Ubuntu server 22.04
- x64

### Your logs

```text
-
```

**Comment by RainerZufall187** [comment]
help
![image](https://github.com/containrrr/watchtower/assets/40537673/b05eaabc-0fcb-41cb-af1c-cc4f08f02aba)

**Comment by piksel** [comment]
It looks like watchtower fails to update itself. Please post the logs to see the reason for why this happens.

**Comment by thomasd3** [comment]
We suddenly have pages of this in the logs and tons of dead containers:

<img width="979" alt="image" src="https://github.com/containrrr/watchtower/assets/14313218/ef61afd6-5b23-4547-91dd-d03d77ccd7eb">

**Comment by piksel** [comment]
@thomasd3 I don't think that is the same issue. You seem to have created a `/config.json` directory in your container root. This can happen if you try to mount a file that doesn't exist as a volume.

**Comment by piksel** [comment]
I have a fix for this, but I cannot fix deploy it due to the docker hub registry server interruption:
https://www.dockerstatus.com/

**Comment by thomasd3** [comment]
That's odd because we had tons of watchtower containers.
If anyone's curious, you can kill them all at once with:
```
docker kill $(docker ps | grep watchtower | sed 's/ .*//')
```

PS: you're right, we had this folder and no clue why. Unrelated issue.

**Comment by piksel** [comment]
> That's odd because we had tons of watchtower containers.

Yes, **that** is related!

**Comment by piksel** [comment]
I finally managed to push `v1.6.0` as `:latest` (attempt 60?), which should mitigate this until docker hub is stable at least...

**Comment by JA16122000** [comment]
ould not do a head request for "containrrr/watchtower:latest", falling back to regular pull.
Reason: Get "https://auth.docker.io/token?scope=repository%3Acontainrrr%2Fwatchtower%3Apull&service=registry.docker.io": EOF
Unable to update container "/watchtower": Error response from daemon: Get "https://registry-1.docker.io/v2/": EOF. Proceeding to next.
Could not do a head request for "mrlt8/wyze-bridge:latest", falling back to regular pull.
Reason: Get "https://index.docker.io/v2/": EOF
Unable to update container "/wyze-bridge": Error response from daemon: Get "https://registry-1.docker.io/v2/": EOF. Proceeding to next.
Could not do a head request for "portainer/portainer-ce:latest", falling back to regular pull.
Reason: Get "https://index.docker.io/v2/": EOF
Unable to update container "/portainer": Error response from daemon: Get "https://registry-1.docker.io/v2/": EOF. Proceeding to next.

**Comment by MFYDev** [comment]
Same issue here, just noticed that on dockerhub 1.7.0 is no longer available and bad containers are all over my server. This kind of messed my server up.

**Comment by piksel** [comment]
@JA16122000 docker hub is having service degradation right now:  https://www.dockerstatus.com/

**Comment by JA16122000** [comment]
I'll just have to wait ?

**Comment by coro1404** [comment]
The "self-update" pulled an x86 image instead of arm64

`# docker compose up
[+] Running 2/0
 ✔ Container watchtower-watchtower-1                                                                                                                         Created0.0s 
 ! watchtower The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested 0.0s 
Attaching to watchtower-watchtower-1
watchtower-watchtower-1  | exec /watchtower: exec format error
watchtower-watchtower-1 exited with code 0` 

docker-compose.yml unchanged

**Comment by ilike2burnthing** [comment]
Same issue as was reported 3 weeks ago on `:latest-dev` - https://github.com/containrrr/watchtower/issues/1809.

**Comment by piksel** [comment]
@coro1404 yes, I pushed the v1.6.0 image as the `:latest`, but I could not make it a multi-arch tag unfortunately. I will make a proper fix as soon as the build system is working again (the docker hub being down affects the GitHub builds as well).

**Comment by VrTech** [comment]
Same issue as well

**Comment by piksel** [comment]
v1.7.1 is now released which should fix the issue.

**Comment by coro1404** [comment]
yes fixed with 1.7.1 - thanks a lot!

**Comment by ilike2burnthing** [comment]
Yep, back up and running without duplicates. Thanks.
